app/services/publish_service.rb

```diff
@@ -52,42 +54,32 @@ def set_first_published_at
   end
 
   def publish_new_images
-    return unless current_edition.image_revisions.any?
+    assets = edition.image_revisions.flat_map(&:assets)
 
-    current_edition.image_revisions.each do |image_revision|
-      image_revision.assets.each do |asset|
-        raise "Expected asset to be on asset manager" if asset.absent?
+    assets.each do |asset|
+      raise "Expected asset to be on asset manager" if asset.absent?
+      next unless asset.draft?
 
-        if asset.draft?
-          AssetManagerService.new.publish(asset)
-          asset.live!
-        end
-      end
+      AssetManagerService.new.publish(asset)
+      asset.live!
     end
   end
 
-  def retire_old_images
+  def retire_old_images(live_edition)
     return unless live_edition
 
-    image_revisions = current_edition.image_revisions
-    current_file_revisions = image_revisions.map(&:file_revision)
-    current_revisions_by_image_id = image_revisions.group_by(&:image_id)
-
-    live_edition.image_revisions.each do |revision|
-      next if current_file_revisions.include?(revision.file_revision)
+    live_edition.image_revisions.each do |old_revision|
+      new_revision = edition.image(old_revision.image_id)
```

Comment:
Were you planning to do something different with this? I'm assuming it's slipped through as I can't see this method defined.

Reply:
Oops - sorry about that. Fixed.